File: coding-challenges/cracking-the-coding-interview/problems/01_isUnique.ts
// 1. *Is Unique*:

// Implement an algorithm to determine if a string has all unique characters.
// What if you cannot use additional data structures?

export default function isUnique(str: string): boolean {
    const letter = str.split("");
    for (let i = 0; i < letter.length; i++) {
      for (let j = i + 1; j < letter.length; j++) {
        if (letter[j] === letter[i]) return false;
      }
    }
    return true;
}

File: coding-challenges/cracking-the-coding-interview/problems/01_isUnique.py
"""// Implement an algorithm to determine if a string has all unique characters.
// What if you cannot use additional data structures?"""

def isUnique(stringValue):
    for i in range(len(stringValue)):
        for j in range(i+1,len(stringValue)):
            if (stringValue[i] == stringValue[j]):
               return False
    return True

"""// Implement an algorithm to determine if a string has all unique characters.
// You can use additional data structures
def isUnique1(stringValue):
    values = []
    for i in range(len(stringValue)):
        if stringValue[i] in values:
            return False

        values.append(stringValue[i])
    
    return True
"""